### ISSUE TITLE
How to build it in Android Studio?

### ISSUE THREAD

**Issue by 2fish** [servalproject/serval_chat#1]
After I clone your project from github and build it in Android Studio, it says Error:Execution failed for task ':app:ndkClean'.> A problem occurred starting process 'command 'F:\Android_ADT\adt-bundle-windows-x86_64-20140702\sdk\ndk-bundle/ndk-build''  
How can I build it correctly?

**Comment by lakeman** [comment]
Do you have the ndk installed?

(I don't try to build on windows, so other than that....)

**Comment by 2fish** [comment]
Yes,I use Android Studio buildin Ndk.And I use “gradlew compile --stacktrace" command,it says
java.lang.UnsupportedClassVersionError: com/android/build/gradle/AppPlugin : Unsupported major.minor version 52.0.

**Comment by 2fish** [comment]
I try it on Ubuntu 16.04,then it says can' find libsodium.a .

**Comment by lakeman** [comment]
Right, you have to build that first;
$ cd app/src/main/jni/libsodium
$ ANDROID_NDK_HOME="{PATH}" ./dist-build/android-arm.sh

**Comment by 2fish** [comment]
I do as you said. And it occurs "make-standalone-toolchain.sh used to install a new toolchain into an existing directory. This is not desirable, as it will not clean up any stale files. If you wish to remove the install directory before creation, pass --force.". Even if I pass "--force" after the command, it's still the same. Should I do any configuration before I start building it ?

**Comment by lakeman** [comment]
Looks like we need to push a patch like this to libsodium;

dist-build/android-build.sh
@@ -50,6 +50,9 @@ if [ "$NDK_PLATFORM" != "$NDK_PLATFORM_COMPAT" ]; then
   echo
   echo "Configuring again for platform [${NDK_PLATFORM}]"
   echo
+
-  rm -rf "${TOOLCHAIN_DIR}"
  +
  bash $MAKE_TOOLCHAIN --platform="$NDK_PLATFORM" \
     --arch="$ARCH" --install-dir="$TOOLCHAIN_DIR" || exit 1

(or passing --force...)
